packages/account/src/Components/poi/status/unsupported/unsupported.tsx

```diff
@@ -49,25 +53,79 @@ const Unsupported = ({
     });
 
+    const from_platform = getPlatformRedirect(app_routing_history);
+
+    const onClickRedirectButton = () => {
+        const platform = platforms[from_platform?.ref as keyof typeof platforms];
+        const { is_hard_redirect = false, url = '' } = platform ?? {};
+        if (is_hard_redirect) {
+            window.location.href = url;
+            window.sessionStorage.removeItem('config.platform');
+        } else {
+            routeBackTo?.(from_platform.route);
+        }
+    };
+
+    const status_content = getPOIStatusMessages(
+        manual?.status ?? AUTH_STATUS_CODES.NONE,
+        { needs_poa },
+        !!redirect_button
+    );
+
+    const upload_complete_status_content = getUploadCompleteStatusMessages(
+        AUTH_STATUS_CODES.PENDING,
+        { needs_poa, is_manual_upload: true },
+        !!redirect_button
+    );
+
     if (manual) {
-        if (manual.status === identity_status_codes.pending)
-            return <UploadComplete is_manual_upload needs_poa={needs_poa} redirect_button={redirect_button} />;
-        else if ([identity_status_codes.rejected, identity_status_codes.suspected].includes(manual.status)) {
-            if (!allow_poi_resubmission) return <Limited />;
-        } else if (manual.status === identity_status_codes.verified) {
-            return <Verified needs_poa={needs_poa} redirect_button={redirect_button} />;
-        } else if (manual.status === identity_status_codes.expired) {
-            return <Expired redirect_button={redirect_button} handleRequireSubmission={handleRequireSubmission} />;
+        let content: typeof status_content | typeof upload_complete_status_content = {
+            title: null,
+            icon: '',
+            description: null,
+            action_button: null,
+        };
+        if (manual?.status === AUTH_STATUS_CODES.PENDING) {
+            content = upload_complete_status_content;
+        } else if (
+            manual?.status === AUTH_STATUS_CODES.VERIFIED ||
+            manual?.status === AUTH_STATUS_CODES.EXPIRED ||
+            ([AUTH_STATUS_CODES.REJECTED, AUTH_STATUS_CODES.SUSPECTED].some(status => status === manual.status) &&
+                !allow_poi_resubmission)
+        ) {
+            content = status_content;
         }
+
+        const onClickHandler = () => {
+            if (manual?.status === AUTH_STATUS_CODES.VERIFIED || manual?.status === AUTH_STATUS_CODES.PENDING) {
+                return onClickRedirectButton;
+            } else if (manual?.status === AUTH_STATUS_CODES.EXPIRED) {
+                return handleRequireSubmission;
+            }
+        };
+
+        return (
+            <VerificationStatus
+                icon={content.icon}
+                is_mobile={isMobile()}
+                status_description={content.description}
+                status_title={content.title}
+            >
+                {content.action_button?.({ onClick: onClickHandler(), platform_name: from_platform.name })}
```

Comment:
```suggestion
                {content.action_button?.({ onClick: onClickHandler(), platform_name: from_platform.name })}
```

isn't it better to assign the returned function from `onClickHandler` to a variable than calling the function inside the object which returns a function?

Reply:
I don’t get your point. and then ssigne this variable to this field in this object? what is the difference?

Reply:
Difference is that this function is being called on each render regardless you need it or not and returns a new function